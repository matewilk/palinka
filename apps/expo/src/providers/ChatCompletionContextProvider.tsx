import {
  createContext,
  useState,
  useContext,
  PropsWithChildren,
  Dispatch,
  SetStateAction,
} from "react";

export type Message = {
  role: "system" | "user" | "assistant";
  content: string;
};

export type ChatCompletionContextType = {
  chatCompletion: Message[];
  setChatCompletion: Dispatch<SetStateAction<Message[]>>;
  addMessage: ({ role, content }: Message) => void;
  resetChatCompletion: () => void;
  resetUserPrompt: () => void;
  isAssistant: boolean;
};

export const ChatCompletionContext = createContext<
  ChatCompletionContextType | undefined
>(undefined);

export const ChatCompletionProvider = (props: PropsWithChildren) => {
  const [chatCompletion, setChatCompletion] = useState<Message[]>([]);

  const addMessage = ({ role, content }: Message) => {
    const newMessage: Message = { role, content };
    setChatCompletion((prevChatCompletion) => [
      ...prevChatCompletion,
      newMessage,
    ]);
  };

  const resetChatCompletion = () => {
    setChatCompletion([]);
  };

  const resetUserPrompt = () => {
    setChatCompletion((prevChatCompletion) =>
      prevChatCompletion.filter((message) => message.role === "system"),
    );
  };

  const isAssistant =
    chatCompletion[chatCompletion.length - 1]?.role === "assistant";

  return (
    <ChatCompletionContext.Provider
      value={{
        chatCompletion,
        setChatCompletion,
        addMessage,
        resetChatCompletion,
        resetUserPrompt,
        isAssistant,
      }}
      {...props}
    />
  );
};

export const useChatCompletion = () => {
  const context = useContext(ChatCompletionContext);
  if (!context) {
    throw new Error(
      "useChatCompletion must be used within a ChatCompletionProvider",
    );
  }

  return context;
};
