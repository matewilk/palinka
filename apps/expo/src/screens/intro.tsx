import React, { FC } from "react";
import {
  View,
  SafeAreaView,
  Text,
  TouchableOpacity,
  Image,
} from "react-native";
import { StackNavigationProp } from "@react-navigation/stack";
import Animated, { FadeInUp, FadeInDown } from "react-native-reanimated";

import { ScreenIndicator } from "../components/ScreenIndicator";
import { AuthStackParamList } from "../navigation/AuthStackNavigator";
import { tokens, translate } from "../i18n";

type IntroScreenProps = {
  navigation: StackNavigationProp<AuthStackParamList>;
};

export const IntroScreen: FC<IntroScreenProps> = ({ navigation }) => {
  return (
    <SafeAreaView className="flex-1">
      <Animated.View
        entering={FadeInUp.duration(1000).springify()}
        exiting={FadeInUp.duration(1000).springify()}
        className="flex-1 items-center justify-end"
      >
        <Image
          className="h-64 w-64 rounded-sm"
          source={{ uri: "https://via.placeholder.com/250" }}
        />
      </Animated.View>
      <View className="flex-1 items-center justify-end gap-10 p-5">
        <Animated.Text
          entering={FadeInDown.duration(1000).springify()}
          className="text-2xl font-extrabold"
        >
          {translate(tokens.screens.intro.header)}
        </Animated.Text>
        <Animated.Text
          entering={FadeInDown.delay(200).duration(1000).springify()}
          className="text-xl text-gray-600"
        >
          {translate(tokens.screens.intro.subheader)}
        </Animated.Text>
      </View>
      <Animated.View entering={FadeInUp.delay(400).duration(1000).springify()}>
        <ScreenIndicator count={3} activeIndex={1} />
      </Animated.View>

      <Animated.View
        entering={FadeInDown.delay(600).duration(1000).springify()}
        className="w-full items-center p-5"
      >
        <TouchableOpacity
          className="w-full items-center rounded-xl bg-black p-3"
          onPress={() => navigation.navigate("Login")}
        >
          <Text className="text-lg text-white">
            {translate(tokens.screens.intro.nextBtn)}
          </Text>
        </TouchableOpacity>
      </Animated.View>
    </SafeAreaView>
  );
};
