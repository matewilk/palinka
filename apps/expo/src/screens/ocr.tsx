import React, { useState, useEffect } from "react";
import {
  Text,
  View,
  Image,
  SafeAreaView,
  ScrollView,
  Dimensions,
} from "react-native";
import { DetectDocumentTextCommandOutput } from "@aws-sdk/client-textract";
import type { StackScreenProps } from "@react-navigation/stack";
import Animated, { FadeInUp, FadeInDown } from "react-native-reanimated";
import { OcrStackParamList } from "../navigation/OcrStackNavigator";

interface TextractResultsProps {
  results: DetectDocumentTextCommandOutput;
}

const TextractResults: React.FC<TextractResultsProps> = ({ results }) => {
  const [textLines, setTextLines] = useState<string[]>([]);

  useEffect(() => {
    if (results?.Blocks) {
      const lines = results.Blocks.filter(
        (block) => block.BlockType === "LINE",
      ).map((block) => block.Text || "");
      setTextLines(lines);
    }
  }, [results]);

  return (
    <ScrollView>
      {textLines.map((line, index) => (
        <Text key={index}>{line}</Text>
      ))}
    </ScrollView>
  );
};

const screenHeight = Dimensions.get("window").height;

import { isPortrait } from "../hooks/useImageUpload";

type OcrResultScreenProps = StackScreenProps<OcrStackParamList, "Result">;

export const OcrResultScreen = ({ route }: OcrResultScreenProps) => {
  const { image, data } = route.params;

  return (
    <SafeAreaView>
      <View className="h-full p-4">
        <Animated.View
          className="items-center"
          entering={FadeInDown.duration(500).springify()}
        >
          {image && (
            <View className="flex items-center p-1">
              <Image
                source={{ uri: image.uri }}
                style={{
                  aspectRatio: image.width / image.height,
                  height: isPortrait(image)
                    ? screenHeight * 0.2
                    : screenHeight * 0.1,
                }}
                className="w-full bg-primary-lightest"
              />
            </View>
          )}
        </Animated.View>
        <Animated.ScrollView>
          <TextractResults results={data} />
        </Animated.ScrollView>
      </View>
    </SafeAreaView>
  );
};
