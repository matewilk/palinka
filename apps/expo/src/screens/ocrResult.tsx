import React from "react";
import {
  View,
  Image,
  SafeAreaView,
  Dimensions,
  TouchableOpacity,
  Text,
} from "react-native";
import { StackScreenProps } from "@react-navigation/stack";
import Animated, { FadeInDown, FadeInUp } from "react-native-reanimated";
import { SheetManager } from "react-native-actions-sheet";

import { OcrStackParamList } from "../navigation/OcrStackNavigator";
import { isPortrait } from "../hooks/useImageUpload";
import TextractResults from "../components/TextractResults";
import TaskSelectionSheet from "../components/TaskSelectionSheet";

import { Selection } from "../types/app";
import { translate, tokens } from "../i18n";

const selection: Selection[] = [
  {
    title: "Edit text",
    message: {
      role: "user",
      content: "",
    },
  },
  {
    title: "Respond to text",
    message: {
      role: "user",
      content: "",
    },
  },
];

const screenHeight = Dimensions.get("window").height;

type OcrResultScreenProps = StackScreenProps<OcrStackParamList, "Result">;

export const OcrResultScreen = ({
  route,
  navigation,
}: OcrResultScreenProps) => {
  const { image, data } = route.params;

  return (
    <SafeAreaView>
      <View className="h-full p-4 pb-2 pt-2">
        <Animated.View
          className="items-center"
          entering={FadeInDown.duration(500).springify()}
        >
          {image && (
            <View className="flex items-center p-1">
              <Image
                source={{ uri: image.uri }}
                style={{
                  aspectRatio: image.width / image.height,
                  height: isPortrait(image)
                    ? screenHeight * 0.3
                    : screenHeight * 0.2,
                  borderRadius: 8,
                }}
              />
            </View>
          )}
        </Animated.View>
        <View className="my-2 border-b border-gray-300" />
        <TextractResults results={data} />
        <Animated.View
          entering={FadeInUp.delay(300).duration(500).springify()}
          className="w-full items-center justify-end pt-2"
        >
          <TouchableOpacity
            onPress={() => {
              SheetManager.show("ocr-task-selection");
            }}
            className="w-full items-center justify-center rounded-full bg-primary p-3"
          >
            <Text className="text-lg text-white">
              {translate(tokens.screens.home.selectTaskBtn)}
            </Text>
          </TouchableOpacity>
        </Animated.View>
      </View>
      <TaskSelectionSheet
        sheetId="ocr-task-selection"
        payload={{ navigation: navigation }}
        selection={selection}
        navigationRoute="Prompt"
      />
    </SafeAreaView>
  );
};
