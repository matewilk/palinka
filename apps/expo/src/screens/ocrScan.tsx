import { useState } from "react";
import {
  SafeAreaView,
  ScrollView,
  Dimensions,
  Image,
  View,
  Text,
  TouchableOpacity,
} from "react-native";
import type { StackScreenProps } from "@react-navigation/stack";
import Animated, { FadeInUp, FadeInDown } from "react-native-reanimated";

import Logo from "../../assets/logo.svg";
import { OcrStackParamList } from "../navigation/OcrStackNavigator";
import { translate, tokens } from "../i18n";
import {
  useImageUpload,
  getFirstImage,
  isPortrait,
} from "../hooks/useImageUpload";

const screenHeight = Dimensions.get("window").height;

type OcrScanScreenProps = StackScreenProps<OcrStackParamList, "Scaner">;

export const OcrScanScreen = ({ navigation }: OcrScanScreenProps) => {
  const { image, pickImage, uploadImage, detectText, status } =
    useImageUpload();
  const [hasScanned, setHasScanned] = useState(false);

  const firstImage = getFirstImage(image);

  const isProcessing = status === "SCANNING" || status === "ANALYSING";

  const handleScan = async () => {
    setHasScanned(true);
    if (firstImage) {
      const filename = await uploadImage();
      if (filename) {
        const data = await detectText(filename);
        if (data) {
          navigation.navigate("Result", { image: firstImage, data });
        }
        setHasScanned(false);
      }
    }
  };

  return (
    <SafeAreaView>
      <ScrollView
        contentContainerStyle={{
          flexGrow: 1,
          justifyContent: "space-between",
          flexDirection: "column",
        }}
        className="h-full p-2"
      >
        <Animated.View entering={FadeInDown.duration(500).springify()}>
          <View className="items-center">
            <Logo width={100} height={30} />
          </View>
        </Animated.View>

        <Animated.View
          className="items-center"
          entering={FadeInUp.delay(300).duration(500).springify()}
        >
          {firstImage ? (
            <View className="flex items-center p-1">
              <Image
                source={{ uri: firstImage.uri }}
                style={{
                  aspectRatio: firstImage.width / firstImage.height,
                  height: isPortrait(firstImage)
                    ? screenHeight * 0.6
                    : screenHeight * 0.25,
                }}
                className="w-full rounded-xl bg-primary-lightest"
              />
            </View>
          ) : (
            <Text className="text-center text-xl">
              {translate(tokens.screens.ocrScan.subheader)}
            </Text>
          )}

          <View className="flex items-center p-1">
            <Text className="text-center text-xl">
              {hasScanned ? `${status.toLowerCase()}` : " "}
            </Text>
          </View>
        </Animated.View>

        <Animated.View
          entering={FadeInUp.delay(300).duration(500).springify()}
          className="items-center"
        >
          <Animated.View
            entering={FadeInDown.delay(200).duration(500).springify()}
            className="w-full text-lg"
          >
            <TouchableOpacity
              onPress={pickImage}
              className={`w-full items-center justify-center rounded-full bg-primary p-3 ${
                isProcessing ? "opacity-50" : ""
              }`}
              disabled={isProcessing}
            >
              <Text className="text-lg text-white">
                {translate(tokens.screens.ocrScan.selectImageBtn)}
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={handleScan}
              className={`mt-3 w-full items-center justify-center rounded-3xl p-3 ${
                !firstImage || isProcessing
                  ? "bg-primary opacity-50"
                  : "bg-primary"
              }`}
              disabled={!firstImage || isProcessing}
            >
              <Text className="text-lg text-white">
                {translate(tokens.screens.ocrScan.scanBtn)}
              </Text>
            </TouchableOpacity>
          </Animated.View>
        </Animated.View>
      </ScrollView>
    </SafeAreaView>
  );
};
