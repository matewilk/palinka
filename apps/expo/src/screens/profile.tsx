import React from "react";
import {
  Text,
  SafeAreaView,
  View,
  TextInput,
  TouchableOpacity,
} from "react-native";

export const ProfileScreen = () => {
  return (
    <SafeAreaView>
      <View className="h-full w-full p-4">
        {/* Container for the avatar icon, text, and button */}
        <View className="flex-1 items-center justify-start pt-24">
          <View className="flex h-20 w-20 items-center justify-center rounded-full bg-gray-300">
            <Text className="text-4xl font-bold text-gray-500">P</Text>
          </View>

          <View className="w-full flex-1 items-center justify-center space-y-4 pt-4">
            <View className="w-full items-center space-y-1">
              <Text className="text-center">First Name</Text>
              <TextInput
                className="w-full rounded border border-gray-300 py-2 px-3"
                placeholder="First Name"
              />
            </View>

            <View className="w-full items-center space-y-1">
              <Text className="text-center">Last Name</Text>
              <TextInput
                className="w-full rounded border border-gray-300 py-2 px-3"
                placeholder="Last Name"
              />
            </View>

            <View className="w-full items-center space-y-1">
              <Text className="text-center">Email Address</Text>
              <TextInput
                className="w-full rounded border border-gray-300 py-2 px-3"
                placeholder="Email Address"
              />
            </View>
          </View>

          <View className="w-full flex-1 items-center justify-end pb-4">
            <TouchableOpacity className="w-full items-center rounded bg-red-400 py-3">
              <Text className="text-center font-bold text-white">
                Delete Account
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};
