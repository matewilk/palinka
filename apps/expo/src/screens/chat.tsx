import { useEffect, useState } from "react";
import { View, Text } from "react-native";
import { useIsFocused } from "@react-navigation/native";
import { SafeAreaView } from "react-native-safe-area-context";
import {
  GiftedChat,
  Send,
  Bubble,
  IMessage,
  BubbleProps,
  SendProps,
} from "react-native-gifted-chat";
import { ActivityIndicator } from "react-native";
import Ionicons from "@expo/vector-icons/Ionicons";

import { trpc } from "../utils/trpc";
import { useChatCompletion } from "../providers/ChatCompletionContextProvider";
import { tokens, translate } from "../i18n";
import { TouchableOpacity } from "react-native-gesture-handler";

export const ChatScreen = () => {
  const { chatCompletion, addMessage, isAssistant, getGiftedChatMessages } =
    useChatCompletion();

  const isFocused = useIsFocused();
  const [prompt, setPrompt] = useState("");

  const { mutate, data, isLoading } = trpc.openai.chatCompletion.useMutation();

  // When first screen enters, mutete to send a request to OpenAI
  useEffect(() => {
    if (isFocused && !isAssistant) {
      mutate(chatCompletion, {
        onSuccess: (data) =>
          addMessage({
            role: "assistant",
            content: data.message?.content as string,
          }),
      });
    }
  }, [isFocused, data]);

  // When the user sends a message, mutate to send a request to OpenAI
  const handleSend = (messages: IMessage[] = []) => {
    messages.forEach((message) => {
      addMessage({
        role: "user",
        content: message.text,
      });
      mutate(chatCompletion);
    });
  };

  const renderBubble = (props: BubbleProps<IMessage>) => {
    return (
      <Bubble
        {...props}
        wrapperStyle={{
          left: {
            backgroundColor: "#f0f0f0",
          },
          right: {
            backgroundColor: "#000",
          },
        }}
        textStyle={{
          left: {
            color: "#000",
          },
          right: {
            color: "#fff",
          },
        }}
      />
    );
  };

  const renderSend = (props: SendProps<IMessage>) => {
    return (
      <Send {...props}>
        <TouchableOpacity
          testID="gf-send-btn"
          className="h-full bg-gray-800 py-2 px-4"
          onPress={() => props.onSend?.({ text: prompt.trim() }, true)}
        >
          <Ionicons name="send" size={24} color="white" />
        </TouchableOpacity>
      </Send>
    );
  };

  const renderFooter = () => {
    if (isLoading) {
      return (
        <View
          style={{ margin: 10, alignItems: "center", justifyContent: "center" }}
          className="m-5 justify-center align-middle"
        >
          <ActivityIndicator size="small" color="#000" />
          <Text className="mt-5">
            {translate(tokens.screens.chat.assistantWorking)}
          </Text>
        </View>
      );
    }
    return null;
  };

  return (
    <SafeAreaView className="flex-1">
      <GiftedChat
        messages={getGiftedChatMessages()}
        onSend={(messages) => handleSend(messages)}
        user={{
          _id: 1,
        }}
        renderBubble={renderBubble}
        renderSend={renderSend}
        renderFooter={renderFooter}
        bottomOffset={20}
        text={prompt}
        onInputTextChanged={(text) => setPrompt(text)}
        renderAvatar={() => null}
        placeholder={translate(tokens.screens.chat.inputPlaceholder)}
      />
    </SafeAreaView>
  );
};
