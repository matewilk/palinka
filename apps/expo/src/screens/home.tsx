import { useRef } from "react";
import { SafeAreaView, View, Text, TouchableOpacity } from "react-native";
import type { StackScreenProps } from "@react-navigation/stack";
import ActionSheet, {
  ActionSheetRef,
  SheetProps,
  SheetManager,
} from "react-native-actions-sheet";

import { tokens, translate } from "../i18n";
import { useChatCompletion } from "../providers/ChatCompletionContextProvider";
import { Selection } from "../types/app";

const selection: Selection[] = [
  {
    title: translate(tokens.selection.tasks.composeParentNote.title),
    message: {
      role: "system",
      content: translate(tokens.selection.tasks.composeParentNote.content),
    },
  },
  {
    title: translate(tokens.selection.tasks.polishDocument.title),
    message: {
      role: "system",
      content: translate(tokens.selection.tasks.polishDocument.content),
    },
  },
  {
    title: translate(tokens.selection.tasks.createLessonPlan.title),
    message: {
      role: "system",
      content: translate(tokens.selection.tasks.createLessonPlan.content),
    },
  },
];

const SelectionButton = ({
  title,
  onPress,
}: {
  title: string;
  onPress: () => void;
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      className="mt-4 h-10 w-full items-center justify-center rounded-lg border"
    >
      <Text>{title}</Text>
    </TouchableOpacity>
  );
};

const TaskSelectionSheet = ({ sheetId, payload }: SheetProps) => {
  const { addMessage, resetChatCompletion } = useChatCompletion();

  const actionSheetRef = useRef<ActionSheetRef>(null);
  const { navigation } = payload;
  return (
    <ActionSheet ref={actionSheetRef} id={sheetId}>
      <View className="p-4">
        <View>
          {selection.map(({ title, message }) => (
            <SelectionButton
              key={title}
              title={title}
              onPress={() => {
                resetChatCompletion();
                addMessage(message);
                navigation.navigate("Prompt");
              }}
            />
          ))}
        </View>
      </View>
    </ActionSheet>
  );
};

import { MainStackParamList } from "../navigation/MainStackNavigator";

type HomeScreenProps = StackScreenProps<MainStackParamList, "Home">;

export const HomeScreen = ({ navigation }: HomeScreenProps) => {
  return (
    <SafeAreaView>
      <View className="h-full w-full p-4">
        {/* Container for the avatar icon, text, and button */}
        <View className="flex-1 items-center justify-start pt-24">
          <View className="flex h-20 w-20 items-center justify-center rounded-full bg-gray-300">
            <Text className="text-4xl font-bold text-gray-500">P</Text>
          </View>

          {/* Text */}
          <Text className="mt-4 p-4 pb-0 text-lg">
            {translate(tokens.screens.home.helloWithName, { name: "Palinka" })}
          </Text>
          <Text className="mt-4 px-4 text-lg">
            {translate(tokens.screens.home.helloContent)}
          </Text>

          {/* Rounded button */}
          <View className="flex-1 items-center justify-end pb-4">
            <TouchableOpacity
              onPress={() => {
                SheetManager.show("task-selection");
              }}
              className="mt-6 h-10 w-32 items-center justify-center rounded-lg bg-blue-500"
            >
              <Text className="text-white">
                {translate(tokens.screens.home.selectTaskBtn)}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
      <TaskSelectionSheet
        sheetId="task-selection"
        payload={{ navigation: navigation }}
      />
    </SafeAreaView>
  );
};
