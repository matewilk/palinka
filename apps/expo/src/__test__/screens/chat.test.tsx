import React, { ReactNode } from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render, fireEvent } from "@testing-library/react-native";
import { ChatCompletionProvider } from "../../providers/ChatCompletionContextProvider";
import { ChatScreen } from "../../screens/chat";

jest.mock("@clerk/clerk-expo", () => ({
  useAuth: jest.fn().mockReturnValue({ isSignedIn: true }),
  getToken: jest.fn().mockImplementation(() => () => "token"),
}));

jest.mock("../../utils/trpc", () => ({
  trpc: {
    openai: {
      chatCompletion: {
        useMutation: jest.fn().mockImplementation(() => ({
          mutate: () => ({}),
          data: {
            message: {
              role: "assistant",
              content: "Hello World",
            },
          },
          isLoading: false,
        })),
      },
    },
  },
}));

describe("ChatScreen", () => {
  const wrapper = ({ children }: { children: ReactNode }) => (
    <ChatCompletionProvider>
      <NavigationContainer>{children}</NavigationContainer>
    </ChatCompletionProvider>
  );

  it("should render correctly", () => {
    const { toJSON } = render(<ChatScreen />, { wrapper });
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render message content", () => {
    const { getByTestId, getByText } = render(<ChatScreen />, { wrapper });

    const input = getByTestId("auto-expanding-text-input");
    fireEvent.changeText(input, "Hello World");

    expect(getByText("Hello World")).toBeDefined();
  });
});
