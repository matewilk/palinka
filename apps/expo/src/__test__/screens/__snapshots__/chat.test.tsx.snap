// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "height": "100%",
        },
        {
          "width": "100%",
        },
        {
          "flexDirection": "column",
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "flex-start",
            },
            {
              "paddingLeft": 16,
              "paddingRight": 16,
            },
          ]
        }
      >
        <RCTScrollView
          style={
            [
              {
                "marginTop": 8,
              },
              {
                "width": "100%",
              },
            ]
          }
        >
          <View>
            <Text>
              Hello World
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        [
          {
            "height": 56,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginTop": 20,
            },
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "width": "100%",
              },
              {
                "flexDirection": "row",
              },
              {
                "paddingLeft": 16,
                "paddingRight": 16,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "width": "75%",
                },
                {
                  "paddingRight": 4,
                },
              ]
            }
          >
            <TextInput
              multiline={true}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              placeholder="How can I improve the results?"
              style={
                [
                  [
                    {
                      "marginBottom": 8,
                    },
                    {
                      "borderBottomLeftRadius": 4,
                      "borderBottomRightRadius": 4,
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    {
                      "borderBottomWidth": 2,
                      "borderLeftWidth": 2,
                      "borderRightWidth": 2,
                      "borderTopWidth": 2,
                    },
                    {
                      "borderBottomColor": "#6b7280",
                      "borderLeftColor": "#6b7280",
                      "borderRightColor": "#6b7280",
                      "borderTopColor": "#6b7280",
                    },
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                  ],
                  {
                    "height": 5,
                  },
                ]
              }
              testID="auto-expanding-text-input"
              value=""
            />
          </View>
          <View
            style={
              [
                {
                  "width": "25%",
                },
                {
                  "paddingLeft": 8,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#3b82f6",
                  "borderBottomLeftRadius": 4,
                  "borderBottomRightRadius": 4,
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                  "display": "flex",
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontWeight": "600",
                    },
                  ]
                }
              >
                Send
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
